Dockerfile

```diff
@@ -1,15 +1,13 @@
 FROM golang:1.4
 
 ENV CONFIG_PATH /etc/docker/registry/config.yml
-RUN mkdir -pv "$(dirname $CONFIG_PATH)"
-
 ENV DISTRIBUTION_DIR /go/src/github.com/docker/distribution
+ENV GOPATH $DISTRIBUTION_DIR/Godeps/_workspace:$GOPATH
+
 WORKDIR $DISTRIBUTION_DIR
 COPY . $DISTRIBUTION_DIR
-ENV GOPATH $GOPATH:$DISTRIBUTION_DIR/Godeps/_workspace
-
-RUN go install -v ./cmd/registry
-
+RUN make PREFIX=/go clean binaries
+RUN mkdir -pv "$(dirname $CONFIG_PATH)"
 RUN cp -lv ./cmd/registry/config.yml $CONFIG_PATH
```

Comment:
I'm not sure what the best approach here is. This is doing exactly what was there before. Really, we should be configuring with environment variables or make an image based on this image that includes the config.